core/templates/dev/head/services/AssetsBackendApiService.js

```diff
@@ -0,0 +1,96 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to serve as the interface for fetching and uploading
+ * assets from Google Cloud Storage.
+ */
+
+oppia.factory('AssetsBackendApiService', [
+  '$http', '$q', 'GCS_AUDIO_DOWNLOAD_URL_TEMPLATE',
+  'AUDIO_UPLOAD_URL_TEMPLATE', 'UrlInterpolationService',
+  function($http, $q, GCS_AUDIO_DOWNLOAD_URL_TEMPLATE,
+    AUDIO_UPLOAD_URL_TEMPLATE, UrlInterpolationService) {
+    var _loadAudio = function(explorationId, filename,
```

Comment:
I think maybe we can use _fetch... here? Take a look at the naming established in core/templates/dev/head/domain/collection/CollectionRightsBackendApiService.js, for example, re the difference between "fetch" and "load".